plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java

```diff
@@ -209,6 +241,98 @@ public void addTrustedPaths(SnykTrustedFoldersParams param) {
 				.map(s -> s.trim()).distinct().collect(Collectors.joining(File.pathSeparator)));
 	}
 
+	@JsonNotification(value = "$/snyk.scan")
+	public void snykScan(ScanParams param) {
+		var cacheKey = new ScanInProgressKey(param.getFolderPath(), param.getProduct());
+		var scanStateHashMap = ScanState.getInstance().getScanInProgress();
+		switch(param.getStatus()) {
+		case "inProgress":
+			scanStateHashMap.put(cacheKey, true);
+			// Show scanning view state
+			break;
+		case "success":
+			scanStateHashMap.put(cacheKey, false);
+			// Show scanning finished state
+			break;
+		case "error":
+			scanStateHashMap.put(cacheKey, false);
+			// Show error state
+			break;
+		}	
+	}
```

Comment:
suggestion: at a certain point we could think about embedding this kind of code as a subrepo, as IntelliJ uses something similar.